src/structures/BaseGuildVoiceChannel.js

```diff
@@ -68,8 +68,11 @@ class BaseGuildVoiceChannel extends GuildChannel {
    */
   get joinable() {
     if (!this.viewable) return false;
-    if (!this.permissionsFor(this.client.user).has(Permissions.FLAGS.CONNECT, false)) return false;
-    return true;
+    const permissions = this.permissionsFor(this.client.user);
```

Comment:
This needs a check for nullability, `permissionsFor(User)`'s return is nullable.

As for why the old code didn't account for that... it's a bug and would have thrown an error if `guild.me` was not cached.